### PR TITLE
Custom theming fixes for Adwaita 3.20/GTK 3.20

### LINKED ISSUE #1135: Text in the search dialog or the toolbar search field becomes invisible
Sometimes the text I'm searching for becomes invisible in the input field. I can see that it's there if I select it (press Ctrl+A); otherwise, I cannot see it.
Presumably, it's happening if geany cannot find the text entered.

My system is: Arch Linux; geany 1.28 (built against gtk2); gtk2 2.24.30; gtk3 3.20.6; glib 2.48.1; Cinnamon 3.0.5.

Could be the same issue as #1101 or a different one.

### PR DESCRIPTION
**Pull request by b4n**
Fixes #1135, and ~~may help for~~ #1101 ~~(?)~~

This also restores the custom selection color there too that got lost with GTK 3.20, and adds it to GTK2 too (for some reason it never was there).

### CODE REVIEW

File: data/geany.gtkrc [hunk 1]
```diff
@@ -20,6 +20,16 @@ widget "GeanyDialogSearch.*.geany-search-entry-no-match" style "geany-monospace"
 style "geany-search-entry-no-match-style" {
 	base[NORMAL] = "#ffff66666666"
 	text[NORMAL] = "#ffffffffffff"
+	base[SELECTED] = "#777711111111"
+	# try and remove the entry background image on pixmap engine so that our
+	# background color is visible, and we don't end up with white text on white
+	# background (workaround for Adwaita 3.20).
```
Comment (elextr) [inline data/geany.gtkrc]: Why does this affect 3.20? I thought gtk3 didn't use the .gtkrc file?

Comment (b4n) [inline data/geany.gtkrc]: *Adwaita* 3.20.

**Comment by b4n** [comment]
Maybe we should add this to a 1.28.1 release, as quite a few people are likely to suffer from that Adwaita issue?

**Comment by codebrainz** [comment]
Would it be worth reporting the breakage to Adwaita also?

**Comment by codebrainz** [comment]
As for 1.28.1 release, I have no opinion (except that Adwaita is ugly and I don't understand why anyone would use it :)

**Comment by b4n** [comment]
> Would it be worth reporting the breakage to Adwaita also?

I doubt they would care, but it'd be great if someone tried :)

**Comment by dmaphy** [comment]
Anybody knows how widely spread is the Adwaita theme outside from Fedora? If it affects Fedora only, I can back port the patch and a Geany release could wait then.

**Comment by elextr** [comment]
Well, as its the default theme for Gnome I guess it is gonna be available on all Gnome distros, even if the distro defaults to its own hacked theme.

**Comment by b4n** [comment]
@dmaphy as @elextr said, it's the default GNOME theme so it's likely to affect a lot of people.  For example, it affects current Debian testing/unstable.
OTOH, some widespread distros like Ubuntu ship with a custom theme by default, so they won't be affected by the same issues.  Who knows if there are other there, though…

But yeah, I think it's likely to have a large impact, hence why I'm considering a point release might be a good idea.  Either way you may want to backport the fix in Fedora until we release it, though, your call.

**Pull request merged**